src/components/Avatar.tsx

```diff
@@ -0,0 +1,119 @@
+import React, {useEffect, useState} from 'react';
+import {StyleProp, View, ViewStyle} from 'react-native';
+import {ValueOf} from 'type-fest';
+import useNetwork from '@hooks/useNetwork';
+import * as ReportUtils from '@libs/ReportUtils';
+import {AvatarSource} from '@libs/UserUtils';
+import * as StyleUtils from '@styles/StyleUtils';
+import useTheme from '@styles/themes/useTheme';
+import useThemeStyles from '@styles/useThemeStyles';
+import CONST from '@src/CONST';
+import {AvatarType} from '@src/types/onyx/OnyxCommon';
+import Icon from './Icon';
+import * as Expensicons from './Icon/Expensicons';
+import Image from './Image';
+
+type AvatarProps = {
+    /** Source for the avatar. Can be a URL or an icon. */
+    source?: AvatarSource;
+
+    /** Extra styles to pass to Image */
+    imageStyles?: ViewStyle[];
+
+    /** Additional styles to pass to Icon */
+    iconAdditionalStyles?: StyleProp<ViewStyle>;
+
+    /** Extra styles to pass to View wrapper */
+    containerStyles?: StyleProp<ViewStyle>;
+
+    /** Set the size of Avatar */
+    size?: ValueOf<typeof CONST.AVATAR_SIZE>;
+
+    /**
+     * The fill color for the icon. Can be hex, rgb, rgba, or valid react-native named color such as 'red' or 'blue'
+     * If the avatar is type === workspace, this fill color will be ignored and decided based on the name prop.
+     */
+    fill?: string;
+
+    /** A fallback avatar icon to display when there is an error on loading avatar from remote URL.
+     * If the avatar is type === workspace, this fallback icon will be ignored and decided based on the name prop.
+     */
+    fallbackIcon?: AvatarSource;
+
+    /** Denotes whether it is an avatar or a workspace avatar */
+    type?: AvatarType;
+
+    /** Owner of the avatar. If user, displayName. If workspace, policy name */
+    name?: string;
+};
+
+function Avatar({
+    source,
+    imageStyles = [],
+    iconAdditionalStyles = [],
+    containerStyles = [],
+    size = CONST.AVATAR_SIZE.DEFAULT,
+    fill,
+    fallbackIcon = Expensicons.FallbackAvatar,
+    type = CONST.ICON_TYPE_AVATAR,
+    name = '',
+}: AvatarProps) {
+    const theme = useTheme();
+    const styles = useThemeStyles();
+    const [imageError, setImageError] = useState(false);
+
+    useNetwork({onReconnect: () => setImageError(false)});
+
+    useEffect(() => {
+        setImageError(false);
+    }, [source]);
+
+    if (!source) {
+        return null;
+    }
+
+    const isWorkspace = type === CONST.ICON_TYPE_WORKSPACE;
+    const iconSize = StyleUtils.getAvatarSize(size);
+
+    const imageStyle = imageStyles?.length ? [StyleUtils.getAvatarStyle(size), imageStyles, styles.noBorderRadius] : [StyleUtils.getAvatarStyle(size), styles.noBorderRadius];
+
+    const iconStyle = imageStyles?.length ? [StyleUtils.getAvatarStyle(size), styles.bgTransparent, imageStyles] : undefined;
+
+    const iconFillColor = isWorkspace ? StyleUtils.getDefaultWorkspaceAvatarColor(name).fill : fill ?? theme.icon;
```

Comment:
```suggestion
    const iconFillColor = isWorkspace ? StyleUtils.getDefaultWorkspaceAvatarColor(name).fill : fill || theme.icon;
```

The default value of fill is `''` which is not nullish and in that case we will never use `theme.icon`.

We should either:
1. Remove the default value
2. Use the logical or

Reply:
Could you check that part again? I think it works as is. Since fill is set as optional and in the current code there isn't a default value provided it will work in it's current implementation.